### PR TITLE
Fix Databricks pandas types bug

### LINKED ISSUE #681: Type merge error in Databricks source
The following error occurs when trying to upload a file to databricks:
image

It is probably due to the `object` type in the DataFrame. This type allows the rows of a column to contain mixed values, which can cause this error.

### PR DESCRIPTION
**Pull request by djagoda881**
## Summary

Closes #681. Added a fix inside the `create table from_pandas()` function to change the data types in rows of pandas.object columns to string.

## Importance

In certain cases, the lack of a change will result in an error in uploading data to databricks

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [] updates docstrings for any new functions or function arguments (if appropriate)
- [x] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `Genesys` source to library.
 - Fixed a bug in `Databricks.create_table_from_pandas()`. The function that converts column names to snake_case was not used in every case. (#672)
 - Added `howto_migrate_sources_tasks_and_flows.md` document. This document will assist the DEs with the viadot 1 -> viadot 2 migration process.
-
+- Fixed a bug in `Databricks.create_table_from_pandas()`. The `object` type in the DataFrame allows to store mixed values in the column which was causing the error. Added changing type of cloumn content to string. (#681)
```
Comment (Trymzet) [inline CHANGELOG.md:21]: It's not clear to me what it means; whether you added a param to do that or it's done under the hood.

I would put it like that to make it clear this is done automatically: "DataFrame columns of type `object` are now automatically converted to `str`"

**Review by Trymzet — CHANGES_REQUESTED** [review]

File: tests/unit/test_databricks.py [hunk 2]
```diff
@@ -272,6 +275,20 @@ def test_snakecase_column_names(databricks):
     databricks.drop_schema(TEST_SCHEMA)
 
 
+def test_mixed_values_df(databricks):
```
[the remark this answers is not in the document]
Comment (djagoda881) [reply tests/unit/test_databricks.py]: [✅ Updated test name in test_databricks.py](https://github.com/dyvenia/viadot/pull/682/commits/1d0fde0e825c83c183b2208dafbed839641f8cc5)

File: CHANGELOG.md (hunk 1 above)
Comment (djagoda881) [reply CHANGELOG.md]: [📝 Corrected entry in CHANGELOG.md](https://github.com/dyvenia/viadot/pull/682/commits/00d676c9368d78b5a7abbbf9fc6d013a690ed853)

File: viadot/sources/databricks.py [hunk 3]
```diff
@@ -246,6 +248,8 @@ def create_table_from_pandas(
                 Defaults to 'fail'.
             snakecase_column_names (bool, optional): Whether to convert column names to snake case.
                 Defaults to True.
+            cast_df_columns (bool, optional): Whether to cast columns dtypes.
```
Comment (Trymzet) [inline viadot/sources/databricks.py:251]: This doesn't really say anything. Which columns are casted? To which types? Why?

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,7 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `Genesys` source to library.
 - Fixed a bug in `Databricks.create_table_from_pandas()`. The function that converts column names to snake_case was not used in every case. (#672)
 - Added `howto_migrate_sources_tasks_and_flows.md` document. This document will assist the DEs with the viadot 1 -> viadot 2 migration process.
-
+- Fixed a bug in `Databricks.create_table_from_pandas()`. DataFrame columns are now automatically converted using the `utils._cast_df_cols`. (#681)
```
Comment (Trymzet) [inline CHANGELOG.md:21]: You missed out the key details, and now it's not clear neither what the exact issue was, nor what the solution is. You need to mention the `object` and `str` words somewhere in there.

File: viadot/utils.py [hunk 5]
```diff
@@ -207,15 +207,19 @@ def _cast_df_cols(df):
     datetime_cols = (col for col, dtype in df.dtypes.items() if dtype.kind == "M")
     bool_cols = (col for col, dtype in df.dtypes.items() if dtype.kind == "b")
     int_cols = (col for col, dtype in df.dtypes.items() if dtype.kind == "i")
+    object_cols = (col for col, dtype in df.dtypes.items() if dtype.kind == "O")
```
Comment (Trymzet) [inline viadot/utils.py:210]: Pls add a test for this

File: viadot/utils.py [hunk 6]
```diff
@@ -207,15 +207,19 @@ def _cast_df_cols(df):
...
 
     for col in datetime_cols:
         df[col] = df[col].dt.strftime("%Y-%m-%d %H:%M:%S+00:00")
 
     for col in bool_cols:
-        df[col] = df[col].astype(pd.Int64Dtype())
+        df[col] = df[col].astype(int)
```
Comment (Trymzet) [inline viadot/utils.py:216]: Why did you change this?

File: viadot/utils.py [hunk 7]
```diff
@@ -207,15 +207,19 @@ def _cast_df_cols(df):
...
 
     for col in int_cols:
-        df[col] = df[col].astype(pd.Int64Dtype())
+        df[col] = df[col].astype(int)
```
Comment (Trymzet) [inline viadot/utils.py:219]: Why is this changed?

**Review by Trymzet — CHANGES_REQUESTED** [review]
